test/integration/acceptance/grantable_permissions_fixture.cpp

```diff
@@ -0,0 +1,222 @@
+  const auto account_id = account_name + "@" + kDomain;
+  return TxBuilder()
+      .creatorAccountId(permitee_account_id)
+      .createdTime(getUniqueTime())
+      .setAccountQuorum(account_id, quorum)
+      .quorum(1)
+      .build()
+      .signAndAddSignature(permitee_key)
+      .finish();
+}
+
+shared_model::proto::Transaction
+GrantablePermissionsFixture::permiteeSetAccountDetail(
+    const shared_model::interface::types::AccountNameType
+        &permitee_account_name,
+    const shared_model::crypto::Keypair &permitee_key,
+    const shared_model::interface::types::AccountNameType &account_name,
+    const shared_model::interface::types::AccountDetailKeyType &key,
+    const shared_model::interface::types::AccountDetailValueType &detail) {
+  const auto permitee_account_id = permitee_account_name + "@" + kDomain;
+  const auto account_id = account_name + "@" + kDomain;
+  return TxBuilder()
+      .creatorAccountId(permitee_account_id)
+      .createdTime(getUniqueTime())
+      .setAccountDetail(account_id, key, detail)
+      .quorum(1)
+      .build()
+      .signAndAddSignature(permitee_key)
+      .finish();
+}
+
+shared_model::proto::Transaction
+GrantablePermissionsFixture::addAssetAndTransfer(
+    const shared_model::interface::types::AccountNameType &creator_name,
+    const shared_model::crypto::Keypair &creator_key,
+    const shared_model::interface::types::AccountNameType &amount,
+    const shared_model::interface::types::AccountNameType &receiver_name) {
+  const auto creator_account_id = creator_name + "@" + kDomain;
+  const auto receiver_account_id = receiver_name + "@" + kDomain;
+  const auto asset_id =
+      integration_framework::IntegrationTestFramework::kAssetName + "#"
+      + kDomain;
+  return TxBuilder()
+      .creatorAccountId(creator_account_id)
+      .createdTime(getUniqueTime())
+      .addAssetQuantity(asset_id, amount)
+      .transferAsset(
+          creator_account_id, receiver_account_id, asset_id, "", amount)
+      .quorum(1)
+      .build()
+      .signAndAddSignature(creator_key)
+      .finish();
+}
+
+shared_model::proto::Transaction
+GrantablePermissionsFixture::transferAssetFromSource(
+    const shared_model::interface::types::AccountNameType &creator_name,
+    const shared_model::crypto::Keypair &creator_key,
+    const shared_model::interface::types::AccountNameType &source_account_name,
+    const std::string &amount,
+    const shared_model::interface::types::AccountNameType &receiver_name) {
+  const auto creator_account_id = creator_name + "@" + kDomain;
+  const auto source_account_id = source_account_name + "@" + kDomain;
+  const auto receiver_account_id = receiver_name + "@" + kDomain;
+  const auto asset_id =
+      integration_framework::IntegrationTestFramework::kAssetName + "#"
+      + kDomain;
+  return TxBuilder()
+      .creatorAccountId(creator_account_id)
+      .createdTime(getUniqueTime())
```

Comment:
Three lines where you set creator and time are repeated in every function, don't we have something in acceptance fixture for that, like baseTx()? If not please add it to reduce duplication

Reply:
Thanks!